### PR TITLE
Check task is correctly added on parallel `TaskService#addTask` calls

### PR DESCRIPTION
**Pull request by mcornaton**

### CODE REVIEW

**Review by jbern0rd — APPROVED** [review]
Thanks

File: src/test/java/com/iexec/core/task/TaskServiceRealRepositoryTest.java [hunk 1]
```diff
@@ -86,20 +84,32 @@ void shouldAddTaskASingleTime() {
...
                 throw new RuntimeException(e);
             }
-        });
+        }).collect(Collectors.toList());
+
+        // Check one execution has added the task,
+        // while the other one has failed.
+        final Optional<Task> task1 = results.get(0);
+        final Optional<Task> task2 = results.get(1);
+        assertThat(task1.isEmpty() || task2.isEmpty()).isTrue();
```
Comment (jeremyjams) [inline src/test/java/com/iexec/core/task/TaskServiceRealRepositoryTest.java:105]: Aren't you looking for a XOR `^` instead?

Comment (mcornaton) [reply src/test/java/com/iexec/core/task/TaskServiceRealRepositoryTest.java]: You're right! I have looked for `assertThatExactlyOneIsTrue` or some similar method, but couldn't get any result. Thanks for the XOR idea, it's a good amelioration.
Please see https://github.com/iExecBlockchainComputing/iexec-core/pull/547/commits/467da7da0063895bd69eee35067341714b21d409.

Comment (mcornaton) [reply src/test/java/com/iexec/core/task/TaskServiceRealRepositoryTest.java]: Based on a suggestion from @jbern0rd, I've refactored this part. We can now add more than 2 tasks at the same time. This helps to prove the robustness of our code. Please see https://github.com/iExecBlockchainComputing/iexec-core/pull/547/commits/929b3fc1e85380142c43078fbec00e670360eb21.

**Review by jeremyjams — APPROVED** [review]
Thanks :+1:

**Pull request merged**